### ISSUE TITLE
Gradle Kotlin DSL

### ISSUE THREAD

**Issue by forresthopkinsa** [bmuschko/gradle-izpack-plugin#11]
Hi, there's no clear way to get the plugin configuration closures to work with the Gradle Kotlin DSL. Any thoughts?

**Comment by forresthopkinsa** [comment]
Okay, I've figured it out.

```kotlin
buildscript {
    dependencies {
        classpath("com.bmuschko:gradle-izpack-plugin:2.1")
    }
}

apply {
    plugin("com.bmuschko.izpack")
}

dependencies {
    add("izpack", "org.codehaus.izpack:izpack-ant:5.1.2")
}

tasks {
    "installer" {
        dependsOn("install")
        dependsOn("izPackCreateInstaller")
    }
}

configure<IzPackPluginConvention> {
    installFile = file("src/main/resources/installer/install.xml")
    baseDir = file("./")
}
```

**Comment by bmuschko** [comment]
The issue isn't really the plugin functionality AFAIK. It's more about the general documentation of the Kotlin DSL by Gradle. At the moment there's almost nothing.

The code snippet you posted above pretty much applies to all plugins that are used from the Kotlin DSL. Feel free to post a pull request with documentation for this plugin if you think it might be useful for other users. Happy to integrate your changes!

**Comment by forresthopkinsa** [comment]
Okay, I've found a much improved method for this, I'd like to record it here:

```kotlin
plugins {
    id("com.dorongold.task-tree") version "1.3"
}

dependencies {
    izpack("org.codehaus.izpack:izpack-ant:5.1.2")
}

val install by tasks.getting
val izPackCreateInstaller by tasks.getting

val installer by tasks.creating {
    dependsOn(install)
    dependsOn(izPackCreateInstaller)
}

cargo {
    installFile = file("src/main/resources/installer/install.xml")
    baseDir = file("./")
}
```

The only thing out of place here is the `cargo` block. I assume it's due to [this line](https://github.com/bmuschko/gradle-izpack-plugin/blob/1afd58864753c6ef058c43153ecfac34aceaf7f7/src/main/groovy/com/bmuschko/gradle/izpack/IzPackPlugin.groovy#L36), but it sure did take me a while to find the right way of doing it.

I think for now this issue will probably be enough to assist anyone trying to track down a solution to this.

**Comment by bmuschko** [comment]
Ha, thanks for catching that. The `cargo` block is correct code-wise. However, that should actually be `izpack`. This might have also caused confusion for other users when configuring the plugin. I am going to correct it the code soon and will let you know.

**Comment by forresthopkinsa** [comment]
No problem :+1:

**Comment by bmuschko** [comment]
I released a new version 3.0 that uses the extension concept. You'd have to use `configure<IzPackPluginExtension>` instead.
